src/App.tsx

```diff
@@ -1,54 +1,128 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { useCallback, useRef, useState } from 'react';
 import './App.scss';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [visibleList, setVisibleList] = useState(false);
+
+  const { name, born, died } = selectedPerson || {};
+
+  const textField = useRef<HTMLInputElement>(null);
+
+  // #region Query
+  const applyQuery = useCallback(debounce(setAppliedQuery, 500), []);
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setVisibleList(true);
+  };
+
+  // #endregion
+
+  const filteredPeople = peopleFromServer.filter(
```

Comment:
better to add hook `useMemo` for cache the calculated value until something changes in the array of dependencies